dvc/remote/base.py

```diff
@@ -516,6 +520,16 @@ def _save(self, path_info, checksum):
             return
         self._save_file(path_info, checksum)
 
+    def _handle_transfer_exception(
+        self, from_info, to_info, exception, operation
+    ):
+        if isinstance(exception, OSError) and exception.errno == 24:
+            raise TooManyOpenFilesError(exception)
```

Comment:
I think it is better to handle this in main.py , as it might raise elsewhere too. E.g. on status even.

Reply:
So what I mean is that in main.py we have a few `try:excepts` for any CLI command. I would put an except there for this and just would print a proper error. What do you think?

Reply:
Ok, as @pared noted, this is not as trivial, because we catch Exception in upload/download, so it won't hit main.py...

Reply:
Btw, this changes the flow from what we've had before. As previously it would catch and return 1 and not re-raise. Not sure about the consequences.